portal/auth.py

```diff
@@ -0,0 +1,49 @@
+import functools
+
+from flask import (
+    Blueprint, flash, g, redirect, render_template, request, session, url_for
+)
+from werkzeug.security import check_password_hash, generate_password_hash
+
+from . import db
+bp = Blueprint("auth", __name__)
+
+@bp.route('/', methods=('GET', 'POST'))
+def login():
+    if request.method == 'POST':
+        email = request.form['email']
+        password = request.form['password']
+
+        cur = db.get_db().cursor()
+        error = None
+        cur.execute(
+            'SELECT * FROM users WHERE email = %s', (email,)
+        )
+        user = cur.fetchone()
+
+        if user is None:
+            error = 'Incorrect email.'
+        elif not check_password_hash(user['password'], password):
+            error = 'Incorrect password.'
```

Comment:
Make sure you are following OWASP authentication guidelines by providing generic error messages.

Reply:
saw a fix for this in #6 , my bad